### PR TITLE
[OFFAPPS-981] Deploying requirements does not break existing ticket links

### PR DESCRIPTION
**Pull request by vaevictis**
## Jira story
[This way please](https://zendesk.atlassian.net/browse/OFFAPPS-981)

## Description
This PR follows the reverted deploy to production of the Linked Ticket App that broke connection between tickets created in the settings-based era.

The crux of how it works is as follows:
1. Checks if there's an `ancestry_field` setting.
2. If so, checks if there's a value in the associated custom field.
3. If so, ancestry value is pulled from that custom field, and is used for retrieving the linked ticket. This case represents a link between tickets created before deployment of the new app.
4. If there's no value in that custom field, the requirement field newly created is used instead. This case represents when the app would have been installed in the past, but the particular ticket loaded "right now" didn't have a linked ticket created with the old app.
5. If there's no ancestry field setting, requirement field is also used. This case represents the app was not installed before deployment of the requirements-based version.

## Notes
This PR also fixes hiding both settings and requirements `Linked Data` fields, thus fixing [this JIRA story](https://zendesk.atlassian.net/browse/OFFAPPS-982).

## Risk
Medium. The previous attempt at deploying the Linked Ticket App to Production broke existing links for our customers. This PR fixes it, but there could be other side effects. That being said, we now have a V1 branch including a `requirements` file that is ready to re-deploy if needed.

### CODE REVIEW

File: src/javascript/app.js [hunk 1]
```diff
@@ -105,32 +105,54 @@ var App = {
     this.currentUser = currentUserData.currentUser
   },
 
-  setAncestryFieldId: function (requirementData) {
-    this.ancestryFieldId = requirementData.requirement_id
+  setAncestryFieldId: function (ancestryFieldId) {
+    this.ancestryFieldId = ancestryFieldId
```
Comment (ocke) [inline src/javascript/app.js]: Image of Yaktocat

Comment (vaevictis) [reply src/javascript/app.js]: Ok, got it @ocke.

File: src/javascript/app.js [hunk 2]
```diff
@@ -295,9 +313,9 @@ var App = {
     let assignee = data.users.find(user => {
       return user.id === data.ticket.assignee_id
     })
-
     let customField = data.ticket.custom_fields.find(field => {
-      return field.id === this.ancestryFieldId
+      // FIXME: In some cases, one is a string, the other is a number
```
Comment (vaevictis) [inline src/javascript/app.js]: parseInt should solve the FIXME issue

File: src/javascript/app.js [hunk 3]
```diff
@@ -654,21 +672,20 @@ var App = {
...
+        this.zafClient.invoke(`ticketFields:custom_field_${fieldData['requirement:linked_data'].requirement_id}.hide`)
       }
-      return this.zafClient.invoke(`${customFieldName}.hide`)
     })
   },
 
-  getAncestryValue: function () {
-    let customField = `ticket.customField:custom_field_${this.ancestryFieldId}`
+  getAncestryValue: function (ancestryFieldId) {
+    let customField = `ticket.customField:custom_field_${ancestryFieldId}`
```
Comment (ocke) [inline src/javascript/app.js]: `this.ancestryFieldId`?

File: src/javascript/app.js (hunk 2 above)
Comment (vaevictis) [reply src/javascript/app.js]: done

File: src/javascript/app.js (hunk 3 above)
Comment (vaevictis) [reply src/javascript/app.js]: Forgot to fix this one. Will be added to whatever next PR I create on this app!

**Pull request merged**